src/platform/Ameba/ConfigurationManagerImpl.cpp

```diff
@@ -152,6 +152,30 @@ CHIP_ERROR ConfigurationManagerImpl::GetPrimaryWiFiMACAddress(uint8_t * buf)
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR ConfigurationManagerImpl::GetUniqueId(char * buf, size_t bufSize)
+{
+    ReturnErrorCodeIf(bufSize != 32, CHIP_ERROR_INVALID_MESSAGE_LENGTH);
+
+#ifdef CHIP_DEVICE_CONFIG_UNIQUE_ID
+    ReturnErrorCodeIf(bufSize < sizeof(CHIP_DEVICE_CONFIG_UNIQUE_ID), CHIP_ERROR_BUFFER_TOO_SMALL);
+    strcpy(buf, CHIP_DEVICE_CONFIG_UNIQUE_ID);
+    return CHIP_NO_ERROR;
+#else
+    char temp[32];
```

Comment:
memset 0 since the loop below will go through the entire buffer.

Reply:
This has been modified, thanks!